### DISCUSSION TITLE
Customize properties Values

### DISCUSSION THREAD

**Discussion by Jimmy-Ahmed** [AutoFixture/AutoFixture#1408] [Q&A] [answered]
Hi,

I have this code

```
T Generate<T>()
{
            fixture.Customize<DeploymentDetails>(tt =>
                                                     tt.With(x => x.Abc, "dsfsf564564")
                                                     .With(x => x.Def, "ddds")
                                                     .With(x => x.Tyr, "weewwe")
                                                     .With(x => x.Nhgf, "aaddw"));
}
```

What is the best way to send With as a parameter maybe something like

As Tuple
Generate((x => x.Abc, "dsfsf564564"), (x => x.Def, "dsfsf564564"), (x => x.Tyr, "dsfsf564564"), (x => x.Nhgf, "dsfsf564564"));

I tried to use .Do(action) but it did not work since it is not meant to assign values.

**Comment by aivascu** [comment, marked as answer]
@Jimmy-Ahmed there is no built-in way to implement this using AutoFixture, but I think you might be able to pass an array of delegates that can "pipe" the With expressions.
What exactly are you trying to accomplish? Feels like you actually want to use the [object builder](https://github.com/AutoFixture/AutoFixture/wiki/Cheat-Sheet#set-property).

```cs
var mc = fixture.Build<DeploymentDetails>()
                .With(x => x.Abc, "dsfsf564564")
                .With(x => x.Def, "ddds")
                .With(x => x.Tyr, "weewwe")
                .With(x => x.Nhgf, "aaddw")
                .Create();
```
